### PR TITLE
Implement formula parsing.

### PR DESCRIPTION
**Pull request by WildCard65**
Also ported float.inc to new syntax.

### CODE REVIEW

File: translations/core.phrases.txt [hunk 1]
```diff
@@ -83,6 +83,17 @@
...
 	
+	"Unknown Variable"
+	{
+		"#format"		"{1:s}"
+		"en"			"Unknown variable {1}"	
+	}
+	
+	"Unknown Error"
+	{
+		"en"			"Unknown error occurred parsing formula"
```
Comment (50Wliu) [inline translations/core.phrases.txt]: `while parsing formula`?

Comment (WildCard65) [inline translations/core.phrases.txt]: Derp, I'll fix that in the morning alone with compile errors

**Comment by asherkin** [comment]
What's the use case for this? Why does it need to be in SourceMod Core?

**Comment by WildCard65** [comment]
So plugins don't have to re-invent a math parser for usage(like FF2 and VSH have their own) while at same time allow for way to have custom variables be used for other plugins in parsing.

**Comment by dvander** [comment]
I like the idea of a simple eval-string tool, but I'd like to understand more about the plugins that need this.

I do think it's worth making it work the same way the language does, i.e., "5+6*3" or something should evaluate as you'd expect it to in SourcePawn. It's not that difficult with only 5-6 operators, you can either do recursive-descent or something like http://en.wikipedia.org/wiki/Shunting-yard_algorithm is also very simple.

**Comment by asherkin** [comment]
Plugins can provide natives and the extension API exists if it has to be C++, I really don't think we need this in core. I can't see it being useful for more than a tiny handful of plugins (well, 1, exactly). Maybe you don't want to maintain this in FF2, but that doesn't mean punting that burden off to SM.

**Comment by WildCard65** [comment]
@dvander FF2 takes a string from a boss keyvalue file and then parses it to a float(rounded to int later) atm ff2 hardcodes the max complexity of a formula to 128 where this uses ke::Vector. Yes SM exposes ADT Array but it uses CellArray which last I heard it's going to be replaced in the near future right?

**Pull request closed without merging**